### PR TITLE
refactor: remove `ContractCallError`

### PR DESCRIPTION
**Pull request by jakmeier**
The indirection from `FunctionCallError` to `ContractCallError`
to `FunctionCallErrorSer` seems to be obsolete after the latest
rounds of refactoring. Let's remove it.

### CODE REVIEW

**Comment by jakmeier** [comment]
another one for #7818 

@matklad I am not sure if we have any reasons to care about preserving what exact `FunctionCallErrorSer` we produce. If not, I think we can avoid converting errors to stings here:

https://github.com/near/nearcore/pull/7881/files#diff-20827cd3bea08c28cc7c00c44e31d4a416dc2028261161a63491149f7f25994aR303-R316

**Comment by matklad** [comment]
>If not, I think we can avoid converting errors to stings here:

Conversion to strings is somewhat intentional: the *structured* errors are essentially legacy errors: we don't want to tie our DB schema to the shape of random error types in Rust. Ideally, what we store in db is

```rust
struct Error {
  code: u16,
  message: String,
}
```


The current `FunctionCallSer` is not quite that yet, but that's where the idea of stringifying comes from.

File: runtime/near-vm-errors/src/lib.rs [hunk 1]
```diff
@@ -298,6 +297,27 @@ pub enum InconsistentStateError {
...
+                FunctionCallErrorSer::ExecutionError(outer_err.to_string())
+            }
+            // TODO: consider using FunctionCallErrorSer::LinkError
+            FunctionCallError::LinkError { msg } => {
+                FunctionCallErrorSer::ExecutionError(format!("Link Error: {}", msg))
+            }
+            // TODO: consider using FunctionCallErrorSer::WasmTrap
+            FunctionCallError::WasmTrap(ref _e) => {
+                FunctionCallErrorSer::ExecutionError(outer_err.to_string())
+            }
```
Comment (matklad) [inline runtime/near-vm-errors/src/lib.rs:316]: Yeah, so this is not TODO, but rather intentional design. The "structured" varians for FunctionCallErrorSer are only there to deserialize legacy error, all new data shoudl not depend on details of representation of errors.

**Review by matklad — APPROVED** [review]
LGTM with TODOs replaced with "yes, this is how it should look"

**Comment by jakmeier** [comment]
> > If not, I think we can avoid converting errors to stings here:
> 
> Conversion to strings is somewhat intentional: the _structured_ errors are essentially legacy errors: we don't want to tie our DB schema to the shape of random error types in Rust. Ideally, what we store in db is
> 
> ```rust
> struct Error {
>   code: u16,
>   message: String,
> }
> ```
> 
> The current `FunctionCallSer` is not quite that yet, but that's where the idea of stringifying comes from.

Ah, I see. That makes sense now!

I removed the TODOs and added some comments that this is deliberate and that we shouldn't use the old errors for new data.

**Pull request merged**